Makefile

```diff
@@ -70,7 +70,9 @@ pylint:
 
 .PHONY: test
 test:
-	@$(PYTHON) -m pytest -v $(TEST)
+	# for circleCI, path must match what is there
+	@mkdir -p test-reports/pytest
+	@$(PYTHON) -m pytest -v $(TEST) --junitxml=test-reports/pytest/results.xml
```

Comment:
It's a bit odd to write a filename `test-report/pytest` to two different files. It is better to control it all in `.circleci/config.yml`. How about this instead?
```
- run: make -p test-reports/pytest
- run: make test PYTHON=/python3.6/bin/python TEST=--junitxml=test-reports/pytest/results.xml
```

Reply:
That looks sensible, thanks